Fronter.NET/Services/ConverterLauncher.cs

```diff
@@ -115,16 +119,109 @@ internal class ConverterLauncher {
 
 		await process.WaitForExitAsync();
 		timer.Stop();
-
+		
 		if (process.ExitCode == 0) {
 			logger.Info($"Converter exited at {timer.Elapsed.TotalSeconds} seconds.");
 			return true;
 		}
-
-		logger.Error("Converter error! See log.txt for details.");
-		logger.Error("If you require assistance please upload log.txt to forums for a detailed postmortem.");
+		
 		logger.Debug($"Converter exit code: {process.ExitCode}");
+		logger.Error("Converter error! See log.txt for details.");
+		if (SentrySdk.IsEnabled) {
+			// When Sentry is enabled, every event is reported but log and save are only uploaded if the user consents.
+			var saveUploadConsent = await MessageBoxManager.GetMessageBoxStandard(
+				title: "Save upload consent",
+				text: "Would you like the application to automatically upload your save file to our error database, " +
+				      "in order to help us fix this issue?",
+				ButtonEnum.OkCancel,
+				Icon.Question
+				).ShowWindowDialogAsync(MainWindow.Instance);
+			if (saveUploadConsent == ButtonResult.Ok) {
+				AttachLogAndSaveToSentry(config);
+				SentrySdk.ConfigureScope(scope => {
+					scope.SetTag("logProvided", "true");
+				});
+			} else {
+				SentrySdk.ConfigureScope(scope => {
+					scope.SetTag("logProvided", "false");
+				});
+			}
+
+			try {
+				SendMessageToSentry(process.ExitCode);
+			} catch (Exception e) {
+				logger.Warn($"Failed to send message to Sentry: {e.Message}");
+			}
+		} else {
+			logger.Error("If you require assistance please visit the converter's forum thread for a detailed postmortem.");
+		}
 		return false;
 	}
+
+	private static async void AttachLogAndSaveToSentry(Configuration config) {
+		SentrySdk.ConfigureScope(scope => scope.AddAttachment("log.txt"));
+		
+		var saveLocation = config.RequiredFiles.FirstOrDefault(f => f.Name == "SaveGame")?.Value;
+		if (saveLocation is null) {
+			return;
+		}
+
+		Directory.CreateDirectory("temp");
+		
+		// Create zip with save file.
+		var dateTimeString = DateTime.Now.ToString("yyyy-MM-dd_HH-mm-ss");
+		var archivePath = $"temp/SaveGame_{dateTimeString}.zip";
+		using (var zip = ZipFile.Open(archivePath, ZipArchiveMode.Create)) {
+			zip.CreateEntryFromFile(saveLocation, new FileInfo(saveLocation).Name);
+		}
+
+		var archiveSize = new FileInfo(archivePath).Length; // Size in bytes.
+		if (archiveSize <= 19 * 1024 * 1024) {
```

Comment:
Every so often a log file gets absurdly large. Why not measure the log file size and use that to determine the limit?

Reply:
Makes sense. Now I check whether the save archive is <= `20 MB - <save size> - <arbitrary 0.5 MB for base Sentry request>.